src/store/actions/types.ts

```diff
@@ -27,42 +30,58 @@ export interface AddUserMessage {
 }
 
 export interface AddResponseMessage {
-  type: typeof ADD_NEW_RESPONSE_MESSAGE,
-  text: string
+  type: typeof ADD_NEW_RESPONSE_MESSAGE;
+  text: string;
 }
 
 export interface ToggleMsgLoader {
-  type: typeof TOGGLE_MESSAGE_LOADER
+  type: typeof TOGGLE_MESSAGE_LOADER;
 }
 
 export interface AddLinkSnippet {
-  type: typeof ADD_NEW_LINK_SNIPPET,
-  link: Link
+  type: typeof ADD_NEW_LINK_SNIPPET;
+  link: LinkParams;
 }
 
 export interface RenderCustomComponent {
-  type: typeof ADD_COMPONENT_MESSAGE,
-  component: ElementType,
-  props: any,
-  showAvatar: boolean
+  type: typeof ADD_COMPONENT_MESSAGE;
+  component: ElementType;
+  props: any;
+  showAvatar: boolean;
 }
 
 export interface DropMessages {
-  type: typeof DROP_MESSAGES
+  type: typeof DROP_MESSAGES;
 }
 
 export interface HideAvatar {
-  type: typeof HIDE_AVATAR,
-  index: number
+  type: typeof HIDE_AVATAR;
+  index: number;
+}
+
+export interface DeleteMessages {
+  type: typeof DELETE_MESSAGES;
+  count: number;
+  id?: string;
 }
 
 export interface SetQuickButtons {
-  type: typeof SET_QUICK_BUTTONS,
-  buttons: Array<QuickButton>
+  type: typeof SET_QUICK_BUTTONS;
+  buttons: Array<{ label: string, value: string | number }>;
+}
+
+export interface SetBadgeCount {
+  type: typeof SET_BADGE_COUNT;
+  count: number;
+}
+
+export interface MarkAllMessagesRead {
+  type: typeof MARK_ALL_READ;
 }
 
 export type BehaviorActions = ToggleChat | ToggleInputDisabled | ToggleMsgLoader;
 
-export type MessagesActions = AddUserMessage | AddResponseMessage | AddLinkSnippet | RenderCustomComponent | DropMessages | HideAvatar;
+export type MessagesActions = AddUserMessage | AddResponseMessage | AddLinkSnippet | RenderCustomComponent
```

Comment:
I have my doubts about this. What is the advantage of having one type `MessagesActions` that can have lots of interfaces. I think the `|` should be use when we have lots of items that go perfect in 2 or 3 groups of types so we can group them. Here i just see a group of lots of types that each of them represents only 1 action. I think this type MessagesActions should be removed and use the specific types for each action. If you have each specific type for each action the search will be easier

Reply:
The advantage of having this type is to describe the dispatch for each reduce with its action type.

I think that it's not necessary to explicitly type each action creator, that could be refactored.